test/Microsoft.TestPlatform.Extensions.BlameDataCollector.UnitTests/BlameCollectorTests.cs

```diff
@@ -142,6 +142,63 @@ public void TriggerSessionEndedHandlerShouldGetDumpFileIfProcDumpEnabled()
 
             // Setup
             this.mockProcessDumpUtility.Setup(x => x.GetDumpFile()).Returns(this.filepath);
+            this.mockBlameReaderWriter.Setup(x => x.WriteTestSequence(It.IsAny<List<TestCase>>(), It.IsAny<string>()))
+                .Returns(this.filepath);
+
+            // Raise
+            this.mockDataColectionEvents.Raise(x => x.TestHostLaunched += null, new TestHostLaunchedEventArgs(this.dataCollectionContext, 1234));
+            this.mockDataColectionEvents.Raise(x => x.TestCaseStart += null, new TestCaseStartEventArgs(new TestCase()));
+            this.mockDataColectionEvents.Raise(x => x.SessionEnd += null, new SessionEndEventArgs(this.dataCollectionContext));
+
+            // Verify GetDumpFiles Call
+            this.mockProcessDumpUtility.Verify(x => x.GetDumpFile(), Times.Once);
+        }
+
+        /// <summary>
+        /// The trigger session ended handler should not dump files if proc dump was enabled and test host did not crash
+        /// </summary>
+        [TestMethod]
+        public void TriggerSessionEndedHandlerShouldNotGetDumpFileIfNoCrash()
+        {
+            // Initializing Blame Data Collector
+            this.blameDataCollector.Initialize(
+                this.GetDumpConfigurationElement(),
+                this.mockDataColectionEvents.Object,
+                this.mockDataCollectionSink.Object,
+                this.mockLogger.Object,
+                this.context);
+
+            // Setup
+            this.mockProcessDumpUtility.Setup(x => x.GetDumpFile()).Returns(this.filepath);
+            this.mockBlameReaderWriter.Setup(x => x.WriteTestSequence(It.IsAny<List<TestCase>>(), It.IsAny<string>()))
+                .Returns(this.filepath);
+
+            // Raise
+            this.mockDataColectionEvents.Raise(x => x.TestHostLaunched += null, new TestHostLaunchedEventArgs(this.dataCollectionContext, 1234));
+            this.mockDataColectionEvents.Raise(x => x.SessionEnd += null, new SessionEndEventArgs(this.dataCollectionContext));
+
+            // Verify GetDumpFiles Call
+            this.mockProcessDumpUtility.Verify(x => x.GetDumpFile(), Times.Never);
+        }
+
+        /// <summary>
+        /// The trigger session ended handler should get dump files if collect dump always was enabled irrespective of completed test case count
+        /// </summary>
+        [TestMethod]
+        public void TriggerSessionEndedHandlerShouldGetDumpFileIfCollectDumpAlwaysIsEnabled()
```

Comment:
We can add UTs for scenarios like: 
1. Dont collect dump when collectdump enabled, collectDumpIOnAbort is true, but process did not crashed
2. When casing order is not maintained lie `DumpTYPE=full`
3. UTs for scenarios when blame parameters are passed from blame commandline arg, they are set in runsettings correctly.
4. UTs when invalid key value pairs are passed from runsettings or via command line to blame.
5. UTs when invalid value is passed in valid key. like `DumpType=blabla`